src/dpdk_msr4_rx.cpp

```diff
@@ -319,49 +301,49 @@ static void lcore_dump()
     }
 
     // Write samples to file
-    //size_fwrite = fwrite(capbuf[0], 1, WRITE_BUFFER_SIZE, file_capture);
     size_fwrite = write(fd, capbuf[0], WRITE_BUFFER_SIZE);
     if (size_fwrite != WRITE_BUFFER_SIZE)
     {
-      
       throw std::runtime_error("Failed to write to file!");
     }
 
     // Close capture file
-    //fclose(file_capture);
     close(fd);
 
     // Increment file index
     idx_file = (idx_file + 1) % 10000;
 
-    // Mark file write as inactive. No need for mutex here, because there
-    // should be plenty of time between access by two different cores
-    fwrite_active[port_id] = false;
+    #endif
     if (!once) {
       rte_log(RTE_LOG_INFO, 0, "%lu - Port %d - First capture buffer written\n", std::time(NULL), port_id);
       once = true;
     }
   }
+  #ifdef FORWARD_ZMQ
+  zmq_close(publisher);
+  zmq_ctx_destroy(context);
+  #endif
+  rte_log(RTE_LOG_INFO, 0, "%lu - Port %d - %d chunks received.\n", std::time(NULL), port_id, n_chunks);
 }
 
 static int launch_lcores(__rte_unused void *dummy)
 {
   // Get lcore id
   int lcore_id = rte_lcore_id();
 
-  if (lcore_id <= 4)
+  if (lcore_id > 0 && lcore_id < NB_PORTS + 1)
   {
-    // Lcores 1-4 handle packet RX
+    // Lcores for packet RX
     lcore_rx();
   }
-  else if (lcore_id >= 5 && lcore_id <= 8)
+  else if (lcore_id > NB_PORTS && lcore_id < 2 * NB_PORTS + 1)
   {
-    // Lcores 5-8 handle file dump
+    // Lcores for file dump
     lcore_dump();
   }
   else
```

Comment:
Do you need the `else` branch if it is empty? Or did you remove the exception by accident?